server/src/main/java/org/opensearch/gateway/PrimaryShardBatchAllocator.java

```diff
@@ -82,6 +84,7 @@ public AllocateUnassignedDecision makeAllocationDecision(ShardRouting unassigned
      * @param allocation    the allocation state container object
      */
     public void allocateUnassignedBatch(List<ShardRouting> shardRoutings, RoutingAllocation allocation) {
+        logger.debug("Starting shard allocation execution for unassigned primary shards: {}", shardRoutings.size());
```

Comment:
Please change this to trace level and if capture the logs before and after the fix ?

Reply:
Done.